File: PullToRefresh/app/src/main/java/com/example/pulltorefresh/MainActivity.java
package com.example.pulltorefresh;

import androidx.appcompat.app.AppCompatActivity;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Collections;

public class MainActivity extends AppCompatActivity {

    //SwipeRefreshLayout pullToRefresh = findViewById(R.id.swipeRefreshLayout);
    ListView l=findViewById(R.id.p);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ArrayList<number> items = new ArrayList<>();
        for (int i = 1; i <= 20; i++) {
            items.add(new number(i));
        }
        //TextView t=findViewById(R.id.text);

        MyAdapter adapter = new MyAdapter(this, 20, items);

        l.setAdapter(adapter);
    }

}

File: ptr/app/src/main/java/com/example/ptr/MainActivity.kt
package com.example.ptr

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.ArrayAdapter
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.card_view.*
import kotlin.String

class MainActivity : AppCompatActivity() {
    var i=1
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    //for in built adapter
        /*val items= arrayOf<Int>(1)
        val adapter= ArrayAdapter<Int>(this,android.R.layout.simple_list_item_1,items)
        list.adapter=adapter*/

      //for using custom view
      var items = arrayOf<Int>(1)
        val adapter=CustomAdapter(this,20,items,refresh)

        list.adapter=adapter

    }
}